Load the tips dataset using the seaborn library. To do so, you need to import the seaborn library and then use the load_dataset() function, as shown in the following code:

In [1]:
import seaborn as sns
tips = sns.load_dataset('tips')

As we can see from the preceding code, after importing the library, a nickname is given to facilitate its use with the script.
 
The load_dataset() function loads datasets from an online repository. The data from the dataset is stored in a variable named tips.

Create a variable, X, to store the features. Use the drop() function to include all of the features but the target, which in this case is named tip. Then, print out the top 10 instances of the variable:

In [2]:
X = tips.drop('tip', axis=1)
X.head(10)

total_bill     sex smoker  day    time  size
0       16.99  Female     No  Sun  Dinner     2
1       10.34    Male     No  Sun  Dinner     3
2       21.01    Male     No  Sun  Dinner     3
3       23.68    Male     No  Sun  Dinner     2
4       24.59  Female     No  Sun  Dinner     4
5       25.29    Male     No  Sun  Dinner     4
6        8.77    Male     No  Sun  Dinner     2
7       26.88    Male     No  Sun  Dinner     4
8       15.04    Male     No  Sun  Dinner     2
9       14.78    Male     No  Sun  Dinner     2

The axis parameter in the preceding snippet denotes whether you want to drop the label from rows (axis = 0) or columns (axis = 1).

Print the shape of your new variable using the X.shape command:

In [3]:
X.shape

(244, 6)

The first value indicates the number of instances in the dataset (244), while the second value represents the number of features (6).

Create a variable, Y, that will store the target values. There is no need to use a function for this. Use indexing to grab only the desired column. Indexing allows you to access a section of a larger element. In this case, we want to grab the column named tip. Then, we need to print out the top 10 values of the variable:

In [4]:
Y = tips['tip']
Y.head(10)

0    1.01
1    1.66
2    3.50
3    3.31
4    3.61
5    4.71
6    2.00
7    3.12
8    1.96
9    3.23
Name: tip, dtype: float64

Print the shape of your new variable using the Y.shape command:

In [5]:
Y.shape

(244,)

The shape should be one-dimensional with a length equal to the number of instances (244).